llvm/test/Transforms/LICM/hoist-binop.ll

```diff
@@ -432,35 +425,10 @@ entry:
 loop:
   %index = phi i64 [ 0, %entry ], [ %index.next, %loop ]
   %step.add = add nsw i64 %index, %c1
-  call void @use(i64 %step.add)
   %index.next = add nuw nsw i64 %step.add, %c2
   br label %loop
 }
 
-;
-; Hoist MUL and drop NSW even if both ops have it.
-define void @mul_no_nsw_2(i64 %c1, i64 %c2) {
```

Comment:
Was removing this test intentional?

Reply:
The reason I removed it is that it no longer tests `hoistBOAssociation`, but it triggers `hoistMulAddAssociation`:

```
define void @mul_no_nsw_2(i64 %c1, i64 %c2) {
entry:
  %factor.op.mul = mul i64 %c1, %c2
  br label %loop

loop:                                             ; preds = %loop, %entry
  %index = phi i64 [ 0, %entry ], [ %step.add.reass, %loop ]
  %step.add.reass = mul i64 %index, %factor.op.mul
  br label %loop
}
```

(the vector version behaves similarly)

Reply:
What if the type's changed to `<2 x i64>`?

Reply:
But `mul_one_use`:

scalar;
```
define void @mul_one_use(i64 %c1, i64 %c2) {
entry:
  %factor.op.mul = mul i64 %c1, %c2
  br label %loop

loop:                                             ; preds = %loop, %entry
  %index = phi i64 [ 0, %entry ], [ %step.add.reass, %loop ]
  %step.add.reass = mul i64 %index, %factor.op.mul
  br label %loop
}
```
vector:
```
define void @mul_one_use(<2 x i64> %c1, <2 x i64> %c2) {
entry:
  %factor.op.mul = mul <2 x i64> %c1, %c2
  br label %loop

loop:                                             ; preds = %loop, %entry
  %index = phi <2 x i64> [ zeroinitializer, %entry ], [ %step.add.reass, %loop ]
  %step.add.reass = mul <2 x i64> %index, %factor.op.mul
  br label %loop
}
```